models/spring-ai-minimax/src/main/java/org/springframework/ai/minimax/MiniMaxChatClient.java

```diff
@@ -0,0 +1,364 @@
+/*
+ * Copyright 2023 - 2024 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.ai.minimax;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.ai.chat.ChatClient;
+import org.springframework.ai.chat.ChatResponse;
+import org.springframework.ai.chat.Generation;
+import org.springframework.ai.chat.StreamingChatClient;
+import org.springframework.ai.chat.metadata.ChatGenerationMetadata;
+import org.springframework.ai.chat.prompt.ChatOptions;
+import org.springframework.ai.chat.prompt.Prompt;
+import org.springframework.ai.minimax.api.MiniMaxApi;
+import org.springframework.ai.minimax.api.common.MiniMaxApiException;
+import org.springframework.ai.model.ModelOptionsUtils;
+import org.springframework.ai.model.function.AbstractFunctionCallSupport;
+import org.springframework.ai.model.function.FunctionCallbackContext;
+import org.springframework.ai.retry.RetryUtils;
+import org.springframework.http.ResponseEntity;
+import org.springframework.retry.support.RetryTemplate;
+import org.springframework.util.Assert;
+import org.springframework.util.CollectionUtils;
+import org.springframework.util.MimeType;
+import reactor.core.publisher.Flux;
+
+import java.util.*;
+import java.util.concurrent.ConcurrentHashMap;
+
+/**
+ * {@link ChatClient} and {@link StreamingChatClient} implementation for
+ * {@literal MiniMax} backed by {@link MiniMaxApi}.
+ *
+ * @author Geng Rong
+ * @see ChatClient
+ * @see StreamingChatClient
+ * @see MiniMaxApi
+ */
+public class MiniMaxChatClient extends
+		AbstractFunctionCallSupport<MiniMaxApi.ChatCompletionMessage, MiniMaxApi.ChatCompletionRequest, ResponseEntity<MiniMaxApi.ChatCompletion>>
```

Comment:
The abstract base class `AbstractFunctionCallSupport` has the method `doChatCompletionStream` that was not implemented.  I'm not sure how you got this code to compile, perhaps there is another commit with that method and some other code as well.  In the meantime, I've added



```
	@Override
	protected Flux<ResponseEntity<MiniMaxApi.ChatCompletion>> doChatCompletionStream(
			MiniMaxApi.ChatCompletionRequest request) {
		throw new RuntimeException("Streaming Function calling is not supported");
	}
```

Reply:
resolved in PR #744